### PR TITLE
#111504508 Enable social authentication (AUTH)

### PR DESCRIPTION
**Pull request by andela-phemmy**
#### What does this PR do?
    This PR configure and initialize social authentication on facebook, google and twitter
#### Description of Task to be completed?
     1. Register and configure app on social media dashboards
     2. Configure passport js to use social strategies
     3. Create MockStrategy for testing
     4. Write test for features
     5. Implement features
#### How should this be manually tested?
      1. Request for social auth admin access
      2. `git clone <url>`  to clone locally
      3. `git checkout ft-enable-social-login`  to go to specific branch
      4. npm install yarn
      5. yarn install
      6. Create .env file and add all your env variables
      7. Configure Postgres locally
      8. yarn migration
      7. `yarn test` to perform unit testing
      8. `yarn start:dev` to start the server 
#### Any background context you want to provide?
      Social authentication provides easy, quick and secure access for users.
#### What are the relevant pivotal tracker stories?
      #111504508
#### Screenshots

screen shot 2018-10-11 at 4 13 05 am
screen shot 2018-10-11 at 4 14 14 am
screen shot 2018-10-11 at 4 14 21 am
screen shot 2018-10-11 at 4 17 44 am

#### Questions:
   None

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -2,4 +2,7 @@
 coverage
 
 # nyc test coverage
-.nyc_output
\ No newline at end of file
+.nyc_output
+
+# unused react file
+/client
```
Comment (KaiserPhemi) [inline .eslintignore]: #### Nitpick
Add a new line here

File: index.js [hunk 2]
```diff
@@ -3,19 +3,28 @@ import express from 'express';
...
+import passportSetup from './server/config/passportConfig';
+import config from './server/config/config';
 
-
-const isProduction = process.env.NODE_ENV === 'production';
-
+dotenv.config();
+passportSetup();
+const isProduction = config.production;
+// console.log('==========>', isProduction);
```
Comment (KaiserPhemi) [inline index.js]: #### Nitpick
Remove unneeded lines of code entirely

File: server/controllers/SocialAuthController.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        user: {
+          email: user.email,
+          username: user.username,
+          bio: user.bio,
+          image: user.imageUrl
+        }
+      };
+
+        // This checks if the user is just signing up on the platform or not
+
```
Comment (KaiserPhemi) [inline server/controllers/SocialAuthController.js]: #### Nitpick
Remove this empty line

File: server/controllers/SocialAuthController.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { User } from '../db/models';
+import { generateToken } from '../helpers/helper';
+
+/**
+     *
```
Comment (KaiserPhemi) [inline server/controllers/SocialAuthController.js]: #### Nitpick
Doc not properly aligned

File: server/helpers/helper.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+/**
+     *
```
Comment (KaiserPhemi) [inline server/helpers/helper.js]: #### Nitpick
Remove blank lines in doc

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

**Comment by andela-phemmy** [comment]
@KaiserPhemi, I already implemented the feedbacks you gave me.

**Review by Victor-Ugwueze — APPROVED** [review]
LGTM

**Review by Mrshinezee — APPROVED** [review]
LGTM

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**